pyfar/dsp/window.py

```diff
@@ -0,0 +1,378 @@
+       [1] Oppenheim, A. V., & Schafer, R. W. (2010). Discrete-time Signal Processing. Pearson.
+    """
+    
+    if length > 0:
+        return np.ones(length)
+    else:
+        raise ValueError('window length has to be >=0 !')
+    
+
+def hann(length):
+    """returns a hann window of given length.
+
+    Parameters
+    ----------
+    length : integer
+        window length in samples
+
+    Returns
+    -------
+    out : np.array()
+        window coefficients
+        
+    Reference
+    ---------
+       [1] Oppenheim, A. V., & Schafer, R. W. (2010). Discrete-time Signal Processing. Pearson.
+    """
+    
+    if length > 0:
+        n = np.arange(0,length)
+        return 0.5-0.5*np.cos(2*np.pi*n/(length-1))
+    else:
+        raise ValueError('window length has to be >=0 !')
+  
+    
+def hamming(length):
+    """returns a hamming window of given length.
+
+    Parameters
+    ----------
+    length : integer
+        window length in samples
+
+    Returns
+    -------
+    out : np.array()
+        window coefficients
+        
+    Reference
+    ---------
+      [1] Oppenheim, A. V., & Schafer, R. W. (2010). Discrete-time Signal Processing. Pearson.
+    """
+    
+    if length > 0:
+        n = np.arange(0,length)
+        return 0.54-0.46*np.cos(2*np.pi*n/(length-1))
+    else:
+        raise ValueError('window length has to be >=0 !')
+
+
+def blackman(length):
+    """returns a blackman window of given length.
+
+    Parameters
+    ----------
+    length : integer
+        window length in samples
+
+    Returns
+    -------
+    out : np.array()
+        window coefficients
+        
+    Reference
+    ---------
+       [1] Oppenheim, A. V., & Schafer, R. W. (2010). Discrete-time Signal Processing. Pearson.
+    """
+    
+    if length > 0:
+        n = np.arange(0,length)
+        return 0.42-0.5*np.cos(2*np.pi*n/(length-1)) + 0.08*np.cos(4*np.pi*n/(length-1))
+    else:
+        raise ValueError('window length has to be >=0 !')
+
+
+def bartlett(length):
+    """returns a bartlett window of given length.
+
+    Parameters
+    ----------
+    length : integer
+        window length in samples
+
+    Returns
+    -------
+    out : np.array()
+        window coefficients
+        
+    Reference
+    ---------
+       [1] Oppenheim, A. V., & Schafer, R. W. (2010). Discrete-time Signal Processing. Pearson.
+    """
+    
+    if length > 0:
+        n = np.arange(0,length)
+        firstHalf = 2*n[0:length//2]/(length-1)
+        secondHalf = 2-2*n[length//2:]/(length-1)
+        return np.concatenate((firstHalf, secondHalf))
+    else:
+        raise ValueError('window length has to be >=0 !')     
+
+
+def kaiser(length,A= 90.36969147005445):
+    """returns a kaiser window of given length.
+
+    Parameters
+    ----------
+    length : integer
+        window length in samples
+    A:
+        approximate peak approximation error in dB ([1] formula 7.74)
+        default value equals Kaiser beta = 9 window
```

Comment:
I think `A` is the side lobe attenuation in dB. We could use `A=90` and mention that is equals `beta approximately 9`